static_report/src/components/ComparisonReport/CRTableList/CRTableListColumnDetails.tsx

```diff
@@ -0,0 +1,77 @@
+import { useMemo } from 'react';
+
+import { CRTableListColumnDetail } from './CRTableListColumnDetail';
+import {
+  getIconForColumnType,
+  transformAsNestedBaseTargetRecord,
+} from '../../../utils/transformers';
+import type {
+  ColumnSchema,
+  TableSchema,
+} from '../../../sdlc/single-report-schema';
+
+export function CRTableListColumnDetails({
+  base,
+  target,
+}: {
+  base: TableSchema;
+  target: TableSchema;
+}) {
+  const memoized = useMemo(
```

Comment:
Hmm, is `useMemo` necessary in this case? Would like to learn more from you on this 👨‍🎓 

My understanding is as follows: 

> You may rely on useMemo as a performance optimization, not as a semantic guarantee. [reactjs docs](https://reactjs.org/docs/hooks-reference.html#usememo)

I'm asking because it seems that `useMemo` is helpful only in certain cases, and I don't think the logic here can be that large or complex yet to see the benefits (it can actually be slower at times)

Whether or not you should use useMemo will highly depend on your use case, but with a complexity of < 100, useMemo hardly seems interesting.

> It is worth noting that the initial renders with useMemo take quite a setback in terms of performance. We expected an initial performance loss of around 5–10% consistently, but have found that this highly depends on the data/processing complexity and can even cause 500% performance losses, which is 100x more performance loss than expected.
(source: [article](https://canimerge.com/should-you-really-use-usememo-in-react-lets-find-out/))

Of course, this is about performance optimization, and I'd rather not over-optimize for something when the tool is a double-edged sword (with tradeoffs).

Reply:
Yes, agree with you. I did that just for an experiment and I found the result not ideal (maybe I did it the wrong way), so I will revert this change.